### PR TITLE
Fix `NotImplementedError` when using FloatFormatter with numerical data types during fit due to PyArrow

### LINKED ISSUE #886: [PyArrow] `NotImplementedError` when using `FloatFormatter` with numerical data types during fit.
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* RDT version: 1.12.4
* Python version: Any
* Operating System: Any

### Error Description

An error occurs when using `pyarrow` numerical data types with `FloatFormatter`. Specifically, the following exception is raised:
```python
PyArrow NotImplementedError: mod not implemented.
```

This error arises during the fit process, mainly in two parts of the code:

1. `learn_rounding_digits` - [utils.py](https://github.com/sdv-dev/RDT/blob/c3ee0fbd1cc43d8b2eb2e998284b009e98ee734f/rdt/transformers/utils.py#L244) When applying a `mod` or `%`.
2. `_validate_values_within_bounds` - [numerical.py](https://github.com/sdv-dev/RDT/blob/c3ee0fbd1cc43d8b2eb2e998284b009e98ee734f/rdt/transformers/numerical.py#L107) When applying `mod` or `%`.

### Steps to reproduce

This steps are made with `SDV` in order to test the entire workflow (which also uses `learn_rounding_digits`.

```python
from sdv.datasets.demo import download_demo, get_available_demos
import pyarrow as pa
import pandas as pd

data, metadata = download_demo(
    modality='multi_table',
    dataset_name='airbnb-simplified'
)

data["users"] = data["users"].astype({
    'age': pd.ArrowDtype(pa.int64()),
    'signup_flow': pd.ArrowDtype(pa.int64()),
})
data["users"].head(5)

from sdv.multi_table import HMASynthesizer

synthesizer = HMASynthesizer(metadata)
synthesizer.fit(data)
synthetic_data = synthesizer.sample()
```

### Additional Context
Here is a potential workaround for this issue: https://github.com/pandas-dev/pandas/issues/58723#issuecomment-2113644607

### Expected Behavior
The `fit` method should handle `pyarrow` numerical data types without raising the `NotImplementedError`. Once this is fixed we expect to support `pyarrow` numerical data types across all our libraries.

### PR DESCRIPTION
**Pull request by fealho**
CU-86b2bhp27, Resolve #886.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b2bhp27 RDT - [PyArrow] `NotImplementedError` when using `FloatFormatter` with numerical data types during fit. #886](https://app.clickup.com/t/86b2bhp27)

File: pyproject.toml [hunk 1]
```diff
@@ -58,6 +58,7 @@ test = [
     'rundoc>=0.4.3,<0.5',
     'pytest-subtests>=0.5,<1.0',
     'pytest-runner >= 2.11.1',
+    'pyarrow',
```
Comment (gsheni) [inline pyproject.toml:61]: We should set a minimum version

File: rdt/transformers/numerical.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def _raise_out_of_bounds_error(self, value, name, bound_type, min_bound, max_bou
 
     def _validate_values_within_bounds(self, data):
         if not self.computer_representation.startswith('Float'):
-            fractions = data[~data.isna() & data % 1 != 0]
+            fractions = data[~data.isna() & (data != (data // 1))]
```
Comment (gsheni) [inline rdt/transformers/numerical.py:107]: Can we add unit & integration tests to this and the other function? To verify that we support the numpy data types and pyarrow data types?

File: pyproject.toml (hunk 1 above)
Comment (pvk-developer) [reply pyproject.toml]: Also can we move it as an `extra` requirement and not `test`.
```
[project.optional-dependencies]
copulas = ['copulas>=0.11.0',]
pyarrow = ['pyarrow>=version']
test = [
    'rdt[pyarrow]',
    'rdt[copulas]',

    'pytest>=3.4.2',
    'pytest-cov>=2.6.0',
    'jupyter>=1.0.0,<2',
    'rundoc>=0.4.3,<0.5',
    'pytest-subtests>=0.5,<1.0',
    'pytest-runner >= 2.11.1',
    'pyarrow',
    'tomli>=2.0.0,<3',
]
```

File: tests/unit/transformers/test_utils.py [hunk 3]
```diff
@@ -225,6 +226,18 @@ def test_learn_rounding_digits_less_than_15_decimals():
     assert output == 3
 
 
+def test_learn_rounding_digits_pyarrow():
```
Comment (pvk-developer) [inline tests/unit/transformers/test_utils.py:229]: can we also add one more using `floats` to make sure it actually learns the digits ?

**Comment by pvk-developer** [comment]
I think some tests are still failing @fealho

**Review by pvk-developer — APPROVED** [review]

**Review by gsheni — APPROVED** [review]

**Pull request merged**